hello_world/include/hello-world.hpp

```diff
@@ -1,4 +1,10 @@
 #pragma once
+#include <iostream>
 #include <string>
 
-std::string say_hello();
\ No newline at end of file
+#ifdef USING_CMAKEPUBLIC_
+#include <CMakePublic/cmake_public.hpp>
```

Comment:
Remove the `#ifdef` and move the CMakePublic header to the source file. `#ifdef` like that is useful for optional dependencies; for simplicity let's just make this a required dependency. The header should be in the source file because no part of it is actually used in declaring the public API of CMakeSuperBuild.